File: 백준/Silver/10828. 스택/스택.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Locale;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<String> stackStr = new Stack<>();
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(br.readLine());

        for (int i=0; i<n; i++){
            String input = br.readLine();
            String [] instruction = input.split("\\s");
            switch(instruction[0]){
                case "push":
                    stackStr.push(instruction[1]);
                    break;
                case "top":
                    try{
                        sb.append(stackStr.peek()+"\n");
                    }catch(Exception e){
                        sb.append("-1"+"\n");
                    }

                    break;
                case "size":
                    sb.append(stackStr.size()+"\n");
                    break;
                case "empty":
                    if(stackStr.empty()){
                        sb.append("1"+"\n");
                    }else {
                        sb.append("0"+"\n");
                    }
                    break;
                case  "pop":
                    try{
                        sb.append(stackStr.pop()+"\n");
                    }catch(Exception e){
                        sb.append("-1"+"\n");
                    }
                    break;
            }
        }

        System.out.println(sb);

    }
}

File: 백준/Silver/10828. 스택/스택.cc

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
using namespace std;
int stack[10001];
int top = -1;


int main(){
    char s[10];
    int n,x;

    scanf("%d",&n);
    for (int i = 0; i < n; i++) {
        scanf("%s",s);
        if(strcmp(s,"push")==0){
            scanf("%d",&x);
            stack[++top] = x;
        }else if(strcmp(s,"top")==0){
            if(top < 0){
                printf("%d\n",-1);
            }else{
                printf("%d\n",stack[top]);
            }
        }else if(strcmp(s,"size")==0){
            printf("%d\n",top+1);
        }else if(strcmp(s,"empty")==0){
            if(top < 0){
                printf("%d\n",1);
            }else {
                printf("%d\n",0);
            }
        }else if(strcmp(s,"pop")==0){
            if(top < 0){
                printf("%d\n",-1);
            }else{
                printf("%d\n",stack[top--]);
            }

        }
    }



    return 0;
}